apis/nsm/v1alpha1/nsm_types.go

```diff
@@ -17,33 +17,23 @@ limitations under the License.
 package v1alpha1
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
-
 // NSMSpec defines the desired state of NSM
 type NSMSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
-
-	// nsmgr configs true or false
-	Insecure bool `json:"insecure"`
+	// tag represents the desired nsm version
+	Version             string            `json:"version"`
+	NsmPullPolicy       corev1.PullPolicy `json:"nsmPullPolicy"`
+	Registry            string            `json:"registry,omitempty"`
+	Organization        string            `json:"organization,omitempty"`
+	RegistryMemoryImage string            `json:"registryMemoryImage"`
+	NsmgrImage          string            `json:"nsmgrImage"`
 
 	// Forwarding plane configs
 	ForwardingPlaneName  string `json:"forwardingPlaneName"`
 	ForwardingPlaneImage string `json:"forwardingPlaneImage"`
```

Comment:
Hi @acmenezes, 

1. IMHO RegistryMemoryImage, NsmgrImage, and ForwardingPlaneImage can also be optional parameters. So in the case of using the official images, users don't need to define them either.
2. In case users are building images from the source and store them in their own image registry, "Organization" (and possibly "Registry") can be different for the 3 images. Can we instead define RegistryMemoryImage/NsmgrImage/ForwardingPlaneImage to be a combination of registry, organization, and image name instead? For example, registryMemoryImage = "ghcr.io/networkservicemesh/cmd-registry-memory". So we don't put any constraints on naming the images. What do you think?

Reply:
@ mandydydy I agree. That makes sense. As I said before I was following the recipe I received in helm chart format. I just made the changes. Let me know if was exactly what you had in mind. If so I can proceed to merge this PR. Thanks a lot for your reviews!

Reply:
Yes, this is what I had in mind. But is ForwardingPlane different from the other two components? Does its image need to be required? And also why does its name is configurable while others are just hardcoded?

Reply:
Agree that ForwardingPlan should be set explicitly, as there are different options.
It looks good to me now.